ui/vty-lib/src/Ariadne/UI/Vty/Widget/Repl.hs

```diff
@@ -66,15 +77,23 @@ initReplWidget uiFace langFace historyFace fullsizeGetter =
       , replWidgetParseResult = mkReplParseResult langFace ""
       , replWidgetOut = [OutputInfo ariadneBanner]
       , replWidgetFullsizeGetter = fullsizeGetter
+      , replWidgetPasswordMode = Nothing
       }
 
-    addWidgetChild WidgetNameReplInput $
-      initBaseEditWidget (widgetParentLens replWidgetCommandL) "default" id (Just $ widgetParentGetter spanAttrs) EnterWithBackslash
+    addWidgetChild WidgetNameReplInput $ initBaseEditWidget
+        (widgetParentLens replWidgetCommandL)
+        "default"
+        id
+        (Just $ widgetParentGetter spanAttrs)
+        (Just $ widgetParentGetter charModifier)
+        EnterWithBackslash
```

Comment:
I would add a separate password widget, which would be shown only in the password mode.
Also, it should be automatically focused, when the password is requested.
I suppose, it should be done in the same manner as `MenuWidget`, i.e. defined in `App` and having a separate mode, so that all events are dispatched to it.